source/upgrade/important-upgrade-notes.rst

```diff
@@ -14,14 +14,18 @@ Important Upgrade Notes
   :alt: Available for Mattermost Self-Hosted deployments.
 
 .. important::
-   - Support for Mattermost Server v5.31 `Extended Support Release <https://docs.mattermost.com/upgrade/extended-support-release.html>`_ has come to the end of its life cycle as of October 15, 2021. Upgrading to Mattermost Server v5.37 `Extended Support Release <https://docs.mattermost.com/upgrade/extended-support-release.html>`_ or later is required.
+   - Support for Mattermost Server v5.37 `Extended Support Release <https://docs.mattermost.com/upgrade/extended-support-release.html>`_ is coming to the end of its life cycle in April 15, 2022. Upgrading to Mattermost Server v6.3 `Extended Support Release <https://docs.mattermost.com/upgrade/extended-support-release.html>`_ or later is required.
    - The "version" CLI command will be deprecated in a future release.
    - Upgrading Mattermost can result in an error “ERROR: default for column "column_name" cannot be cast automatically to type jsonb” if you have non-JSON data in a column. This can also happen if the database data has been manipulated by external processes (e.g. plugins) and in cases where improper input data could get silently truncated due to varchar limits. The data in the affected database column has to be manually fixed.
    - `Collapsed Reply Threads <https://docs.mattermost.com/messaging/organizing-conversations.html>`__ is currently in beta. Before enabling the feature, please ensure you are well versed in the `known issues <https://docs.mattermost.com/messaging/organizing-conversations.html#known-issues>`__, particularly relating to database resource requirements and server performance implications. If you cannot easily scale up your database size, or are running the Mattermost application server and database server on the same machine, we recommended waiting to enable Collapsed Reply Threads until it's `promoted to general availability in Q1 2022 <https://mattermost.com/blog/collapsed-reply-threads-ga/>`__. Learn more about these `performance considerations here <https://support.mattermost.com/hc/en-us/articles/4413183568276>`__.
 
 +----------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
 | If you’re upgrading from a version earlier than... | Then...                                                                                                                                                          |
 +====================================================+==================================================================================================================================================================+
+| v6.3                                               | An asynchronous job was added to fix an issue with collapsed reply threads that resulted in unexpected unread channels. The job may result in an increase in     |
```

Comment:
Tweaking the wording for this: @ashishbhate @cwarnermm can you please review?

v6.3 adds an asynchronous job to fix an issue that resulted in channels with join/leave messages being unexpectedly marked as unread when enabling Collapsed Reply Threads. After the upgrade, the job is started automatically when the server restarts and the job executes in the background so the Mattermost server will not experience downtime. We recommend upgrading during off-peak hours as you may experience higher than average server and database resource utilization while the job is in progress. In our testing, the job takes approximately 3 hours per ~4 million rows in the `ChannelMembership` table using a PostgreSQL database.

You may check the progress of the job by executing this database query: `select * from Jobs where type = 'fix_channel_unreads_for_crt' order by lastactivityat desc;`. The data column will display the number of channel memberships fixed (BadChannelMembershipsFixed), the number of channel memberships checked (TotalChannelMembershipsChecked), and the current Channel and User Id pair that is being processed. 

If you are experiencing severely high server and database resource utilization, you may kill the job by executing `XXXXXX` < @ashishbhate 

Once the job is complete, it will not run again and resource usage will return to average levels.

Reply:
@ashishbhate @agnivade @streamer45 do we have any idea what the expected increase in DB and server load will be based on when it was run in community? From the admin perspective I'm sure they'll want to know what to expect if this job will be running along side the server. Especially considering all the performance incidents recently

Reply:
@ashishbhate I know the time estimate is rough, but do we also have an idea of the when the admin should "sound the alarm" if this takes over X hours / Y rows?

Reply:
>If you are experiencing severely high server and database resource utilization, you may kill the job by executing 


First update the Systems table to stop the job from restarting
`INSERT INTO Systems VALUES ('fix_crt_channel_unreads', true);`

Then restart the mattermost application to kill the existing job.


>Do we have any idea what the expected increase in DB and server load will be? Perhaps we can use the datapoint from when it was run in community? From the admin perspective I'm sure they'll want to know what to expect if this job will be running along side the server - especially considering all the performance incidents recently

>I know the time estimate is rough, but do we also have an idea of the when the admin should "sound the alarm" if this takes over X hours / Y rows?


Its hard to measure this because the job depends on a lot of factors like number of channelmember entries, how many are broken and need fixing. For fixing ChannelMembers its also needs to scan the posts table. The performance, and time taken also depends on the server load and specs. During high load times its possible the job will slow down as well. 
I don't think there is a need to "sound the alarm" unless the application performance starts to suffer, however slow the job may be proceeding.

>If the job is killed (or perhaps paused during high usage times), can it be restarted at the point it left off? If so what are the commands to do that?

Restarting the mattermost server will automatically reschedule the job to run after a few minutes, if the `fix_crt_channel_unreads` key is not set to `true` in the Systems table. If that key is set, then deleting it from the Systems table will reschedule the job after a few minutes.

Reply:
Okay, thanks @ashishbhate. I updated the proposed text above. Please re-review and confirm it's accuracy?  https://github.com/mattermost/docs/pull/5344#discussion_r781660452

Reply:
This section will need to be removed. We've reverted the job from master and release-6.3.